pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -838,6 +846,10 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 			name := node.Name
 			node, err = nc.kubeClient.CoreV1().Nodes().Get(ctx, name, metav1.GetOptions{})
 			if err != nil {
+				// If the node no longer exists, delete the corresponding health data from nodeHealthMap.
```

Comment:
I don't think it's necessary here, just remove it uniformly in the infomer's deletion callback function.